pkg/paths/path.go

```diff
@@ -50,3 +52,28 @@ func GetAllParams(r *http.Request) (*AllPathParams, error) {
 
 	return &AllPathParams{Module: mod, Version: version}, nil
 }
+
+// MatchGlobPattern reports whether any path prefix of name
+// matches the pattern (as defined by path.Match)
+func MatchGlobPattern(pattern, name string) (bool, error) {
+	const op errors.Op = "paths.MatchGlobPattern"
+	pCount := strings.Count(pattern, "/")
+	nCount := strings.Count(name, "/")
+
+	for nCount > pCount {
+		i := strings.LastIndex(name, "/")
+		if i == -1 {
+			break
+		}
```

Comment:
Is this block ever possible to be true?

Reply:
As it stands, it shouldn't. I can remove it if needed. It's there as a backup, in case the code ever changes.